include/osmium/thread/pool.hpp

```diff
@@ -162,7 +162,11 @@ namespace osmium {
 
             static constexpr int default_num_threads = 0;
 
-            static Pool& instance() {
+            static std::unique_ptr<Pool> create_instance(int num_threads = default_num_threads) {
+                return std::unique_ptr<Pool>(new Pool(num_threads, detail::get_work_queue_size()));
+            }
```

Comment:
There is really no reason why the constructor can't be public. So please change that.